sensor/debugger/k8s/fake.go

```diff
@@ -436,6 +437,123 @@ func (f *FakeEventsManager) waitOnMode(events []string) error {
 	return nil
 }
 
+// forEachResource executes a given function to all resources of a given kind
+func (f *FakeEventsManager) forEachResource(client reflect.Value, execFunc func(reflect.Value) error) error {
+	returnVals := client.MethodByName("List").Call([]reflect.Value{
+		reflect.ValueOf(context.Background()),
+		reflect.ValueOf(metav1.ListOptions{}),
+	})
+
+	if len(returnVals) != 2 {
+		return fmt.Errorf("expected 2 values from %s. Received: %d", "List", len(returnVals))
+	}
+	errInt := returnVals[len(returnVals)-1].Interface()
+	if errInt != nil {
+		return errInt.(error)
+	}
+
+	errorList := errorhelpers.NewErrorList("for each resource")
+	items := reflect.Indirect(returnVals[0]).FieldByName("Items")
+	if !items.IsValid() {
+		errorList.AddError(errors.New("invalid kind: missing `Items` property"))
```

Comment:
Minor: we could return here and save the `else` block

Reply:
Done